src/evaluators/BinaryExpression.js

```diff
@@ -199,7 +199,14 @@ export function computeBinary(
       try {
         // generate error if binary operation might throw or have side effects
         resultType = getPureBinaryOperationResultType(realm, op, lval, rval, lloc, rloc);
-        return AbstractValue.createFromBinaryOp(realm, op, lval, rval, loc);
+        let result = AbstractValue.createFromBinaryOp(realm, op, lval, rval, loc);
+
+        // "in" and "instanceof" can throw and therefore must result in temporals
```

Comment:
If getPureBinaryOperationResultType does not throw, it should not be the case that the binary operation may throw or have a side effect.